### DISCUSSION TITLE
Filter visibility change triggers columnFilters change

### DISCUSSION THREAD

**Discussion by qwelol** [KevinVandy/material-react-table#199] [Q&A] [answered]
Hi,
I use manual server filtering like in this example: 
https://github.com/KevinVandy/material-react-table/discussions/9#discussioncomment-3061140

And the problem is that changing the visibility of filters causes a change in the state of the component (maybe because MRT creates new columnFilters array? idk ). 
Accordingly, at the same time, a repeated data request occurs . The example behaves exactly the same. You can check it out.
This is clearly not very good behavior. How to fix this?

Thx 

upd: attach img 
![filter](https://user-images.githubusercontent.com/39767122/201380097-b02a7941-1d15-4f93-a8c7-0d560768479c.gif)

**Comment by KevinVandy** [comment, marked as answer]
Fixed this in v1.3.8. The filters were clearing on mount so added an extra check to make sure it only did it on subsequent renders. NOTE: You will still see these extra fetches with React Strict Mode during development, but should be fixed in prod builds
